go/vt/vtgate/engine/delete.go

```diff
@@ -264,6 +240,12 @@ func (del *Delete) execDeleteByDestination(vcursor VCursor, bindVars map[string]
 			BindVariables: bindVars,
 		}
 	}
+	if len(del.Table.Owned) > 0 {
```

Comment:
Move this check inside the function. That way, you won't have to check on every invocation.

Reply:
This check will happen only once for the execDeleteByDestination func call.